components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthUtil.java

```diff
@@ -535,6 +535,44 @@ public static void invokePreRevocationBySystemListeners(AccessTokenDO accessToke
         }
     }
 
+    /**
+     * This will be called before when Tokens Revoked through Listeners directly.
+     *
+     * @param userUUID - UUID of the user.
+     * @param params   - Additional parameters.
+     */
+    public static void invokePreRevocationBySystemListeners(String userUUID, Map<String, Object> params) {
```

Comment:
How will this work for client credential grant tokens where there is no user? OAuth app client secret rotation or app disabling could trigger a revocation of client credential grant tokens as well.

Reply:
Also, the purpose of this method is not clear from its name. Can you explain when this method should be called?

Reply:
The OAuth app related changes are handled with the separate implementation in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2089/files